terra_notebook_utils/drs.py

```diff
@@ -121,6 +127,55 @@ def copy_to_local(drs_url: str,
         logger.info(f"Downloading {drs_url} to {filepath}")
         blob.download_to_file(fh)
 
+def print_bytes(blob, num_bytes: int, buffer: int = MULTIPART_THRESHOLD):
+    """
+    Print data from a file until a certain number of bytes is hit.
+
+    This is done by pulling chunks of the file and printing until bytes is reached.
+
+    :param blob: The Google blob object.
+    :param num_bytes: Number of bytes to print before stopping.
+    :param buffer: Size of the data chunks to grab each time.  The default buffer is 32Mib.
+    """
+    num_bytes_left = num_bytes
+    start = 0
+    end = buffer
+    while True:
+        if num_bytes_left <= buffer:
+            print(str(blob.download_as_bytes(start=start, end=num_bytes), sys.stdout.encoding), end='')
```

Comment:
Does converting to `str` transform the contents?

For instance, I'd be surprised if `headed_binary` contained a decoded string instead of the first 1024 bytes of my DRS URI:
```
tnu drs head -c 1024 drs://crouching-drs-hidden-access > headed_binary
```

Reply:
Hmmm... I should write a test for it, but it should have the same behavior as the actual head function since I use `sys.stdout.encoding` which tells str() how to process it to stdout.

The purpose was to remove the extraneous `b""` wrapper, which I don't feel is good to output, and adds extra characters if piped to a file.

I'll see if there's a better way and what the actual head function does.